File: splitservice/src/main/java/com/smartsplit/splitservice/Model/FriendPayment.java
package com.smartsplit.splitservice.Model;

import java.time.LocalDateTime;

import lombok.Data;

@Data
public class FriendPayment {
    private Friend friend;
    private int totalDebt;
    private boolean hasPaid;
    private LocalDateTime paidAt;
    private String paymentImageLink;
}

File: smartsplit_link/lib/Model/friend_payment.dart
import 'package:smartsplit_link/Model/friend.dart';

class FriendPayment {
  final Friend friend;
  final int totalDebt;
  final bool hasPaid;
  final String? paymentImageLink;
  final DateTime? paidAt;

  FriendPayment({
    required this.friend,
    required this.totalDebt,
    required this.hasPaid,
    this.paymentImageLink,
    this.paidAt,
  });
}
